### ISSUE TITLE
How to improve f1-score?

### ISSUE THREAD

**Issue by Urimons** [sgoldenlab/simba#229]
Hello, I'm using SimBA to classify self-grooming behavior in mice. I have gone through all the stages of the GUI and also "played" with the hyperparameters of the training but did not manage to change the f1 score (did the training multiple times and the results are quite similar).
Attached here is the output of the "grooming classification report".
![Grooming_classification_report](https://user-images.githubusercontent.com/86400083/212531534-a551a63a-b635-4149-94c3-a48306a14544.png)

What should I do to increase the f1-score? preferably through improving the recall score which is lower.
Thanks a lot, 
Uri

**Comment by Urimons** [comment]
No, I did not tried that before, but I did now. the results are interesting - they seem to be the opposite... as the recall is now high and the precision is low (attached image).
I think that is because I don't really understand the meaning of the random undersampling and the sample ratio... what does under sample ratio "1" means actually? and is there another way to do random sampling? I just wrote "random undersample" in the under sample setting...

![Grooming_classification_report](https://user-images.githubusercontent.com/86400083/212704540-81805832-9a16-49ea-92dc-b1b60cda06fe.png)

**Comment by sronilsson** [comment]
`1` means that you will take all of you grooming annotations (say 1000), and get an equal number N  (1000) of random non-grooming annotations and you train the classifier on that data. If you put 1.2 then you will train on  all of your 1000 grooming annotations and 1200 random non-grooming annotations. See if you can find a ratio that produces optimal balance f1.

**Comment by Urimons** [comment]
The precision and recall is changing but the f1 score is around the same value more or less...
![Grooming_classification_report](https://user-images.githubusercontent.com/86400083/212714569-02bbdd25-c9f7-443c-ac7a-15d66eba4d0b.png)
![Grooming_classification_report](https://user-images.githubusercontent.com/86400083/212714639-bc8d91e1-8254-4ff4-8b87-eb8b0fd983d2.png)
![Grooming_classification_report](https://user-images.githubusercontent.com/86400083/212714680-7125822f-ee18-4e73-bffb-e69a6d0cbb74.png)
This is for under sample ratio of 1, 0.5 and 1.5 respectively...

**Comment by Urimons** [comment]
The highest f1-score that I can receive is 0.607 with a discrimination threshold of 0.3.... that's better than what I had before, however I wish I could get a better f1-score, any ideas?
Or, is the f1-score of 0.6 considered good for behavioral classification?

**Comment by sronilsson** [comment]
I can't say of 0.6 is good, I don't know enough about the data. I hate to suggest "annotate more", as it can involve a lot of work. But, I have pasted an answer below I gave on gitter recently. 

```
In general, it is helpful to be selective in the frames that are being annotated (can also save time), rather than annotating indiscriminately: i wrote one tool [HERE](https://github.com/sgoldenlab/simba/blob/master/docs/pseudoLabel.md) called pseudo-labelling that can help, another called advanced labelling [HERE](https://github.com/sgoldenlab/simba/blob/master/docs/advanced_labelling.md). There is also a tool [HERE](https://github.com/sgoldenlab/simba/blob/master/docs/classifier_validation.md#classifier-validation) to help understand what the classifier has trouble with.
```

E.g., the key to annotations is not so much quantity as quality. Is there some way if figuring out through visualizations if there is a specific form of grooming event the classifier get wrong, and adding those frames as correct annotations to the classifier?
